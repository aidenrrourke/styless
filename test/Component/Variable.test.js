import React from 'react';
import renderer from 'react-test-renderer';
import styled from 'styled-components';
import 'jest-styled-components';

test('Takes the color from props', () => {
    const Div = styled.div`
		background-color: chocolate;
		color: @primary;
	`;
    const tree = renderer.create(<Div primary="red"/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Replaces single variable in linear-gradient', () => {
    const Div = styled.div`
		background: linear-gradient(@start 0%, @end 100%);
	`;
    const tree = renderer.create(<Div start="red" end="blue"/>).toJSON();
    expect(tree).toMatchSnapshot();
});

test('Use the color defiened in the component', () => {
    const Div = styled.div`
        @main: palevioletred;
        color: @main;
	`;
    expect(renderer.create(<Div/>).toJSON()).toMatchSnapshot();
});