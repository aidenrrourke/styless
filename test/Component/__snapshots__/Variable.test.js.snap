// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Replaces single variable in linear-gradient 1`] = `
.c0 {
  background: linear-gradient(red 0%,blue 100%);
}

<div
  className="c0"
  end="blue"
  start="red"
/>
`;

exports[`Takes the color from props 1`] = `
.c0 {
  background-color: chocolate;
  color: red;
}

<div
  className="c0"
/>
`;

exports[`Use the color defiened in the component 1`] = `
.c0 {
  color: palevioletred;
}

<div
  className="c0"
/>
`;
